oxide-auth/src/primitives/registrar.rs

```diff
@@ -291,107 +282,35 @@ pub trait PasswordPolicy: Send + Sync {
     fn check(&self, client_id: &str, passphrase: &[u8], stored: &[u8]) -> Result<(), RegistrarError>;
 }
 
-/// Store passwords using `Pbkdf2` to derive the stored value.
-///
-/// Each instantiation generates a 16 byte random salt and prepends this additionally with the
-/// username. This combined string is then used as the salt using the passphrase as the secret to
-/// derive the output. The iteration count defaults to `65536` but can be customized.
-pub struct Pbkdf2 {
-    /// A prebuilt random, or constructing one as needed.
-    random: Option<SystemRandom>,
-    iterations: NonZeroU32,
-}
+/// Store passwords using `Argon2` to derive the stored value.
+#[derive(Clone, Debug, Default)]
+pub struct Argon2 {}
```

Comment:
This should contain a base configuration, or at least a `_private: ()` member such that this structure is not publicly constructible. Otherwise it will not be possible to make any changes to it later.

Reply:
Done